src/codegen_source.rs

```diff
@@ -153,6 +151,15 @@ impl<'a, W: Write> CodeSourceGenerator<'a, W> {
                                 cg!(self, "}}");
                             }
                         }
+                    } else if let Some(ref enum_type) = elem.enum_type() {
+                        // Introduce a temprorary variable to avoid reference aliasing issues
+                        let temp_name = format!("{}_temp", elem.name());
+                        cg!(self, "{} {};", enum_type, temp_name);
+                        self.write_if_else(&format!("!reader.get_{}({})", enum_type, temp_name), &[
+                            "return;"
+                        ], None)?;
+                        cg!(self, "{} = ({}){};", elem.name(), elem.type_(), temp_name);
```

Comment:
Could you use `{} = static_cast<{}>({});` instead? If the compiler still complains, use `{} = reinterpret_cast<{}>({});` instead.